secv_guis/base_widgets.py

```diff
@@ -48,12 +48,16 @@ def __init__(self, label, parent=None,
         self.file_watcher.directoryChanged.connect(
             lambda: self.update_path(self.dirpath))
 
-    def update_path(self, dirname):
+    def update_path(self, dirname,file_filter=None):
         """
         :param str dirname: The new directory path to be listed.
         """
-        file_names = [f for f in os.listdir(dirname)
+        if file_filter is None:
+            file_names = [f for f in os.listdir(dirname)
                       if f.lower().endswith(tuple(self.extensions))]
+        else:
+            file_names = [f for f in os.listdir(dirname)
+                      if f.lower().endswith(tuple(self.extensions)) and f.lower().find(file_filter.lower())!=-1]
```

Comment:
linting: spaces around != operator

Reply:
If a file_filter is given, extensions won't be used. That is OK, but you should also update the docstring for `update_path` explaining `file_filter`, and the docstring for `__init__` explaining the new behaviour for `extensions`

Reply:
fixed later